App/components/common/TranslatedText.js

```diff
@@ -0,0 +1,52 @@
+import React  from 'react';
```

Comment:
It seems that even though traditional Chinese is the app default, both forms of Chinese are treated as "translations" in the code with everything being English by default. I suggest considering a pattern where all languages are treated equally, where none are treated as "translations" over any others. Then maybe in the code we could do something like this wherever there is text (as a super basic high-level example):
```jsx
import i18n from 'some-i18n-helper'; // Either a library or our own

export default function Menu() {
  return (
    <div>
      <MenuItem label={i18n('menu.settings')} onClick={() => {}} />
    </div>
  )
}
```
The `i18n()` method would perform a lookup using the key `menu.settings` which would return the proper value in whichever language the app is currently set to. This way none of the languages are regarded as "translations" and none of them show up by default in the code.

This pattern is pretty common in the industry and works well with any number of languages.

Reply:
After reviewing the code some more I see what you have here is actually quite similar to what I described, so we're on the same page with slightly different interpretations. I still feel though that using English text as a key seems a bit odd, but that's probably just my personal preference. It just makes it feel like the app is written in English to me even though the default language is not English.

Reply:
I chose this way for a few different reasons:
- `<TranslatedText>Hello world</TranslatedText>` is much more declarative, simpler to implement and easier to understand than `<Text>i18n(context.phraseKey)<Text>`
- The number of translated strings is pretty small and since the scope seems pretty solid - I don't see this fact changing in the near future, so breaking things up into scope/contexts doesn't really seem necessary on this small scale
- The app is largely written by english speakers, so it makes sense to have the code written in english. the string 'settings' and the variable `menu.settings` are both still english anyways
- As far as I can tell, there is no concept of capitalization in Chinese, therefore having a key 'settings' for example allows us to easily do something like this:
```
<TranslatedText>SETTINGS</TranslatedText>
<TranslatedText>Settings</TranslatedText>
<TranslatedText>settings</TranslatedText>
```
and have it give us the results we want with in all three languages with only one key: `settings`
This works because <TranslatedText> uses toLowerCases() on the string so we don't need multiple keys for all the different english capitalizations, or we don't have to run a bunch of functions on the string to achieve the capitalizations we want
- If we don't have a key for the string in <TranslatedText> it doesn't error out, it just falls back to the english string, which sticks out like a sore thumb among the Chinese characters

All in all, the scope of the project seems very clear and stable: A bible app that allows you to choose between english and several versions of Chinese. There is really very little translations that need to be done, and we don't need a lot of the same features from a full featured i18n() package (like pluralization, interpolation, date/time) so I think this solutions works pretty great for this use case. It's lightweight, simple, easy to understand, easy to use, and not prone to errors.

_**That said, I understand that it is going against the common pattern of the i18n() function, which is a valid objection with a lot of clout**_.

So if you and/or @clintKunz feel pretty strongly about doing a "proper" i18n() solutions, I will be completely on board with that.

Reply:
I don't have too strong of an opinion for this project, though here are some additional thoughts:
- I disagree that `<TranslatedText>Hello World</TranslatedText>` is easier to understand, because now we're working with what are essentially locale keys as children props which seems odd to me. At first glance this just looks like an element that may have special styling that contains text, but there is no indication that anything special is done with the text itself unless I open that component and study it. In contrast, `<Text>{i18n('some.key')}</Text>` immediately shows that a function is being used, so without even reading any other code I already know that something is changing here, and that I may need to watch out for that. Also, until I read the PR description in detail, the code appeared as if there were places that had not been localized yet:
```jsx
export default function DrawerHeader({ testament, onPress }) {
  return (
    <Header>
      <TitleContainer selected>
        <Title selected>
          {testament === 'old' ? 'Old T.' : 'New T.'}
        </Title>
      </TitleContainer>
    </Header>
  );
}

// Lots of stuff in between here that makes it harder to see at a glance what `Title` is

const Title = styled(TranslatedText)(({ theme, selected = false }) => ({
  //...
}));
```
I almost left comments in several places that looked like they weren't using the translations, but that seemed weird so I kept looking and that's when I read the PR description in detail. The code was misleading without more knowledge.
- I was under the impression that the goal was to hand this project off to a native Chinese speaking team eventually. If so, it would be respectful to make it less English-centric (though admittedly yes, keys may still be English words, but JS itself and most other programming languages also use English terms regularly due to its efficiency, so locale keys are more "codey" and fit in without being natural English, if that makes any sense). Doing it using the key method also makes it a bit more "neutral", so regardless of who is working on the code, it doesn't contain any bias one way or the other. English speakers don't need to understand Chinese, and Chinese speakers don't need to be fluent in English. Additionally, people don't really write code in Chinese, using English variables and keys is pretty normal (`const 個人資料 = {}`, while valid, would be a real pain to work with even for native Chinese speakers); but natural, conversational English text goes a bit beyond that.
- Case/capitalization isn't an issue at all with keys. You just make sure the value associated with the key you are selecting has the case you want, so the end result is the exact same.
- I also don't think we need a full-blown fancy library to accomplish localization either, but I do feel using natural English text as localization keys is a bit of an anti-pattern. Having worked in several apps that support multiple languages, I've never seen natural text used to accomplish localization. Depending on how the project goes, we could in theory end up having a case show up like this:
```json
{
  "Please select the bookmarks you wish to delete:": "請選取您想要刪除的書籤:",
  "Split view mode allows you to read the Bible in multiple languages for comparison and language study.": "對照模式讓您同時閱讀兩種語言來使對照和學語言更輕易."
}
```
In this example, the JSON objects get pretty unwieldy and there is no indication of where these strings actually belong in the app. As the app grows (which it might?) it could get harder to debug or get into for newer developers. If any punctuation, phrasing, or wording changes for the English version of the app, we'll have to update the locale files for all languages rather than just updating the English value.

So once again, I really don't have a real strong opinion, but these are my thoughts about localization in general based on my past experience. What you have totally works and I'm fine with it if we really don't think the app will grow much beyond what we have. I just like to try to plan ahead and follow patterns that are scalable and proven. I'm also a firm believer in YAGNI, but if there are simple things we can do to prevent trouble in the future, I like to follow those patterns.